middleware/index.ts

```diff
@@ -168,7 +175,10 @@ export default async function systemFn(message: any): Promise<any> {
 
     // evergreen:function:invoke includes an extra 'data' level for BinaryHTTP format
     let bodyPayload: any = message['payload'];
-    if ('data' in bodyPayload && 'ce-id' in headers && headers['ce-specversion'] === '0.3') {
+    if (typeof bodyPayload === 'object' &&
+            'data' in bodyPayload &&
+            'ce-id' in headers && 
```

Comment:
ditto here - JS doesn't allow `in` operator on `string` type, only `object` type.

I'd like to refactor middleware to make it more testable - we should have caught this earlier.